cmd/kubeadm/app/preflight/checks.go

```diff
@@ -651,7 +651,10 @@ func (swc SwapCheck) Check() (warnings, errorList []error) {
 	}
 
 	if len(buf) > 1 {
-		return []error{errors.New("swap is supported for cgroup v2 only; the NodeSwap feature gate of the kubelet is beta but disabled by default")}, nil
+		return []error{errors.New("swap is supported for cgroup v2 only; " +
+			"the NodeSwap feature gate of the kubelet is beta and enabled by default; " +
+			"the kubelet must be properly configured to use swap; " +
+			"or disable swap on node")}, nil
```

Comment:
Is there a document link about the NodeSwap feature gate we can add here?

Reply:
@SataQiu 

Yes, I find two links that are related to the swap memory configuration in Kubernetes. 

- https://kubernetes.io/docs/concepts/architecture/nodes/#swap-memory
- (text is out-date, need to update) https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#before-you-begin
  Swap configuration. The default behavior of a kubelet was to fail to start if swap memory was detected on a node. 
  Swap has been supported since v1.22. And since v1.28, Swap is supported for cgroup v2 only; 
  the NodeSwap feature gate of the kubelet is beta but disabled by default.
  ...

Which link is preferred?

Reply:
I'm +1 for https://kubernetes.io/docs/concepts/architecture/nodes/#swap-memory

Reply:
I remove "the NodeSwap feature gate of the kubelet is beta and enabled by default; ". once the feature is GA, we don't need to change warning messages again.

cc @pacoxu @neolit123 @SataQiu

Reply:
once the feature is GA we should remove the kubeadm check entirely.

Reply:
The NodeSwap FG Gate may only mean the feature gate status. 
https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2400-node-swap/README.md#swap-as-the-default
`We will turn the feature on for Beta 2 but the default setting will be NoSwap.`

The KEP will not change the default value of `--fail-swap-on= ` to false.

Reply:
> The KEP will not change the default value of --fail-swap-on=  to false.

if the feature goes GA and swap is well supported, failing when swap is enabled on a node would be odd.

Reply:
@neolit123  After reading the KEP, it seems no plan to change the default value of the `--fail-on-swap` to `false`.

```
Design Details 

We summarize the implementation plan as following:
...
2. Leave the default value of kubelet flag --fail-on-swap to true, to avoid changing default behaviour.
...

GA:
(Tentative.)

Test a wide variety of scenarios that may be affected by swap support.
Remove feature flag.
Remove the Swap Support using Burstable QoS Pods only deprecated in Beta 2.
```

So, we still check the swap and send warning messages if the swap on the node is enabled.

/cc @kannon92 

For confirming whether the ` --fail-on-swap` is `true` or `false` as default value once it is promoted to GA.

Reply:
> if the feature goes GA and swap is well supported, failing when swap is enabled on a node would be odd.如果该功能正式发布并且 swap 得到很好的支持，那么在节点上启用 swap 时失败会很奇怪。

After reading the KEP, I think that even if this feature is officially released, it is not recommended for users to use it unless they really want to use it. My view may be wrong. It is indeed a bit odd.

Reply:
Perhaps because the risks are too numerous and difficult to address, it is difficult to accept `--fail-on-swap=false` as the default value.

https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2400-node-swap#risks-and-mitigations

Reply:
Yes, we will most likely recommend that this feature is opt in for GA. Not sure if we want all kubelet to be swap enabled as there are still a few open questions (best way to configure swap for performance, limiting swap on a user basis, etc).

Reply:
They aren’t hard questions to answer but not sure if we want/expect every admin to also configure swap.

Reply:
Thank you @kannon92 

Any suggestions on my changes? please take another look, thank you. @pacoxu @SataQiu @neolit123

Reply:
continuing to default a flag with a certain value so that the kubelet fails for a supposedly supported feature would be confusing for future and existing users...

does this mean we need to maintain a warning about swap forever in kubeadm?

Reply:
According to https://github.com/kubernetes/kubernetes/pull/125157#discussion_r1618909907, I think so.

Reply:
i recall discussions in https://github.com/kubernetes/kubeadm/issues/2563 about removing the kubeadm preflight check about swap completely.